bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/SurroundWithBracketsStrategy.java

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright (c) ETAS GmbH 2024, all rights reserved.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     ETAS GmbH - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jface.text;
+
+import java.util.Map;
+
+import org.eclipse.swt.SWT;
+
+import org.eclipse.jface.text.source.ISourceViewer;
+
+/**
+ * @since 3.26 This strategy supports surrounding the selected text with similar opening and closing
+ *        brackets when the text is selected and an opening bracket is inserted.
+ */
+public class SurroundWithBracketsStrategy implements IAutoEditStrategy {
```

Comment:
Does this work with block selections, too? E.g., when the user has multiple selections in the current document?

Reply:
@szarnekow please refer video in this [comment](https://github.com/eclipse-platform/eclipse.platform.ui/pull/2051#issuecomment-2426175808). 

Are you also asking for the same thing?

Reply:
No, it's a different thing (which is likely unsupported yet). For example, in a text `abababa` , try to search for `a` and click "Select All", then you get a multi-selection. In such case, the expectation would be that if someone hits `(` the resulting text becomes `(a)b(a)b(a)b(a)`.